util/webhook/webhook.go

```diff
@@ -189,11 +193,31 @@ func affectedRevisionInfo(payloadIf any) (webURLs []string, revision string, cha
 		// See: https://confluence.atlassian.com/bitbucket/event-payloads-740262817.html#EventPayloads-Push
 		// NOTE: this is untested
 		webURLs = append(webURLs, payload.Repository.Links.HTML.Href)
```

Comment:
Everything network-bound in this block should be handled within a context with a reasonable timeout, i.e. fetching the repo creds and then fetching the diffs.

Reply:
Using context with deadline of 10s to complete all callbacks.